### PR TITLE
CLOUD-3245 - [7.2.x] Allow kubernetes to control probe retries; avoid probes taking longer than kubernetes timeout settings

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/CLOUD-3245
Upstream PR: https://github.com/jboss-container-images/jboss-eap-modules/commit/764ae61a0c8e1b90810a8bb714712ebed515c356
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

**Comment by bstansberry** [comment]
I put the hold label on because this needs an upstream pr.

File: os-eap-probes/added/livenessProbe.sh [hunk 1]
```diff
@@ -16,26 +16,60 @@ COUNT=30
...
+  if [ $# -gt 1 ] ; then
+     DEBUG_SCRIPT=$2
+  fi
+
+  if [ $# -gt 2 ] ; then
+     PROBE_IMPL=$3
+  fi
+else
+    # deprecated support for count / sleep / initial sleep of 5s
+    if [ ! -f ${TMPLOC}/probe_deprecation_warning ]; then
```
Comment (bstansberry) [inline os-eap-probes/added/livenessProbe.sh]: This should also test if $# -gt 0 as I think you'll get here with no params set at all.

File: os-eap-probes/added/readinessProbe.sh [hunk 2]
```diff
@@ -10,21 +10,42 @@ COUNT=30
...
+  if [ $# -gt 1 ] ; then
+     DEBUG=$2
+  fi
+
+  if [ $# -gt 2 ] ; then
+        PROBE_IMPL=$3
+  fi
+else
+    # deprecated support for count / sleep
+    if [ ! -f ${TMPLOC}/probe_deprecation_warning ]; then
```
Comment (bstansberry) [inline os-eap-probes/added/readinessProbe.sh]: Same comment as in liveness probe re $# -gt 0

File: os-eap-probes/added/livenessProbe.sh (hunk 1 above)
Comment (bstansberry) [inline os-eap-probes/added/livenessProbe.sh]: This should also test if $# -gt 0 as I think you'll get here with no params set at all.

**Comment by luck3y** [comment]
This is ready to go now, updated to only warn if a parameter has been supplied.

**Comment by luck3y** [comment]
Rebased and force-pushed to pick up the CI / testing fixes in https://github.com/jboss-container-images/jboss-eap-modules/pull/123 with no other changes.

**Pull request merged**